packages/panzer/adapters-stk/src/Panzer_STK_ModelEvaluatorFactory_impl.hpp

```diff
@@ -312,6 +312,47 @@ namespace panzer_stk {
 
     m_eqset_factory = eqset_factory;
 
+        // Setup active parameters
+    /////////////////////////////////////////////////////////////
+
+    std::vector<Teuchos::RCP<Teuchos::Array<std::string> > > p_names;
+    std::vector<Teuchos::RCP<Teuchos::Array<double> > > p_values;
+    std::vector<std::string> tangent_param_names;
+    if (p.isSublist("Active Parameters")) {
+      Teuchos::ParameterList& active_params = p.sublist("Active Parameters");
+
+      int num_param_vecs = active_params.get<int>("Number of Parameter Vectors",0);
+      p_names.resize(num_param_vecs);
+      p_values.resize(num_param_vecs);
+      for (int i=0; i<num_param_vecs; i++) {
+        std::stringstream ss;
+        ss << "Parameter Vector " << i;
+        Teuchos::ParameterList& pList = active_params.sublist(ss.str());
+        int numParameters = pList.get<int>("Number");
+        TEUCHOS_TEST_FOR_EXCEPTION(numParameters == 0,
+                                   Teuchos::Exceptions::InvalidParameter,
+                                   std::endl << "Error!  panzer::ModelEvaluator::ModelEvaluator():  " <<
+                                   "Parameter vector " << i << " has zero parameters!" << std::endl);
+        p_names[i] =
+          Teuchos::rcp(new Teuchos::Array<std::string>(numParameters));
+        p_values[i] =
+          Teuchos::rcp(new Teuchos::Array<double>(numParameters));
+        for (int j=0; j<numParameters; j++) {
+          std::stringstream ss2;
+          ss2 << "Parameter " << j;
+          (*p_names[i])[j] = pList.get<std::string>(ss2.str());
+          ss2.str("");
+
+          ss2 << "Initial Value " << j;
+          (*p_values[i])[j] = pList.get<double>(ss2.str());
+
+          // this is a band-aid/hack to make sure parameters are registered before they are accessed
+          panzer::registerScalarParameter((*p_names[i])[j],*global_data->pl,(*p_values[i])[j]);
+          tangent_param_names.push_back((*p_names[i])[j]);
+        }
+      }
+    }
+
```

Comment:
Had to move this up so the tangent params were available to be passed to the physics blocks. This allows the appropriate tangent gather evaluators to be set up.